pkg/media/manifest_builder.go

```diff
@@ -0,0 +1,177 @@
+			{
+				"label": "c2pa.actions",
+				"data": obj{
+					"actions": []obj{
+						{"action": "c2pa.created"},
+						{"action": "c2pa.published"},
+					},
+				},
+			},
+			{
+				"label": StreamplaceMetadata,
+				"data": obj{
+					"@context": obj{
+						"dc": "http://purl.org/dc/elements/1.1/",
+						"Iptc4xmpExt": "http://iptc.org/std/Iptc4xmpExt/2008-02-29/",
+						"photoshop": "http://ns.adobe.com/photoshop/1.0/",
+						"xmpRights": "http://ns.adobe.com/xap/1.0/rights/",
+					},
+					"dc:creator": streamerName,
+					// TODO: Add the title of the livestream. This should come from the livestream record.
+					"dc:title":   []string{"livestream"},
+					"dc:date":    []string{aqtime.FromMillis(start).String()},
+				},
+			},
+		},
+	}
+
+	// Add database metadata if available
+	if mb.model != nil {
+		metadata, err := mb.model.GetDefaultMetadata(ctx, streamerName)
+		if err != nil {
+			log.Warn(ctx, "ManifestBuilder: failed to retrieve metadata, using defaults", "error", err, "did", streamerName)
+		} else if metadata != nil {
+			streamplaceMetadata, err := metadata.ToStreamplaceDefaultMetadata()
+			if err != nil {
+				log.Warn(ctx, "ManifestBuilder: failed to convert metadata, using defaults", "error", err, "did", streamerName)
+			} else {
+				mani = mb.enhanceManifestWithMetadata(mani, streamplaceMetadata)
+			}
+		}
+	}
+
+
+	// Convert to C2PA manifest
+	manifestBs, err := json.Marshal(mani)
+	if err != nil {
+		return nil, fmt.Errorf("failed to marshal manifest: %w", err)
+	}
+
+	var manifest c2pa.ManifestDefinition
+	if err := json.Unmarshal(manifestBs, &manifest); err != nil {
+		return nil, fmt.Errorf("failed to unmarshal manifest: %w", err)
+	}
+
+	return &manifest, nil
+}
+
+func (mb *ManifestBuilder) enhanceManifestWithMetadata(mani obj, metadata *streamplace.DefaultMetadata) obj {
+	if metadata.ContentRights != nil {
+		// TODO: We are currently validating the creator in the ValidateMP4 function to be the streamer DID
+		// if metadata.ContentRights.Creator != nil {
+		//	mani["assertions"].([]obj)[1]["data"].(obj)["dc:creator"] = *metadata.ContentRights.Creator
+		// }
+
+		// Copyright Notice
+		if metadata.ContentRights.CopyrightNotice != nil {
+			mani["assertions"].([]obj)[1]["data"].(obj)["dc:rights"] = *metadata.ContentRights.CopyrightNotice
+		}
+
+		// Copyright Year
+		if metadata.ContentRights.CopyrightYear != nil {
+			mani["assertions"].([]obj)[1]["data"].(obj)["Iptc4xmpExt:CopyrightYear"] = *metadata.ContentRights.CopyrightYear
+		}
+
+		// Credit Line
+		if metadata.ContentRights.CreditLine != nil {
+			mani["assertions"].([]obj)[1]["data"].(obj)["photoshop:Credit"] = *metadata.ContentRights.CreditLine
+		}
+
+		// Build the license field
+		if metadata.ContentRights.License != nil {
+			// Map internal license codes to known licenses
+			var licenseCodeMap = map[string]string{
+				"place.stream.default.metadata#cc0_1__0": "http://creativecommons.org/publicdomain/zero/1.0/",
```

Comment:
Constants here seems prudent, I'm totally gonna type `place.stream.default.metadata#cc0_1_0` someday and be completely unable to spot the typo